### PR TITLE
SNOW-176143: Dump telemetry to log

### PR DESCRIPTION
**Pull request by sfc-gh-wshangguan**
Dump telemetry data into a separate log file, in which log records won't be masked by secret detector.

### CODE REVIEW

**Comment by sfc-gh-wshangguan** [comment]
Hey @sfc-gh-mkeller, I found that on jenkins we can't successfully picked up `GITHUB_ACTIONS`, is this the correct behavior?

File: tox.ini [hunk 1]
```diff
@@ -29,8 +29,7 @@ passenv =
     SF_PROJECT_ROOT
     cloud_provider
     SF_REGRESS_LOGS
-    ; Github Actions provided environmental variables
-    GITHUB_ACTIONS
```
Comment (sfc-gh-mkeller) [inline tox.ini:33]: Please don't delete this

Comment (sfc-gh-wshangguan) [reply tox.ini:33]: I thought I wrongly added this line in one of my commits?

Comment (sfc-gh-mkeller) [reply tox.ini:33]: We need it for sure, I think I added this while moving to this repo

Comment (sfc-gh-wshangguan) [reply tox.ini:33]: Sure, let's keep it

**Pull request merged**